git-retimestamp

```diff
@@ -0,0 +1,162 @@
+    if num_commits > 9:
+        # Show only the top and bottom 3 commits
+        for commit in commits_list[0:3]:
+            commit_num += 1
+            commit_hash, commit_title = commit.split(' ', 1)
+            sys.stdout.write('    ' + str(commit_num) + '. (\033[34m' +
+                commit_hash + '\033[0m) ' + commit_title)
+
+            if commit_num == 1:
+                sys.stdout.write("    \033[32m<-- old HEAD of '" + branch_name +
+                    "'\033[0m\n")
+            else:
+                sys.stdout.write('\n')
+
+        print "    ..."
+        print "    ... << {} more commits >> ...".format(num_commits - 6)
+        print "    ..."
+
+        commit_num = num_commits - 3
+        for commit in commits_list[-3:]:
+            commit_num += 1
+            commit_hash, commit_title = commit.split(' ', 1)
+            sys.stdout.write('    ' + str(commit_num) + '. (\033[34m' +
+                commit_hash + '\033[0m) ' + commit_title + '\n')
+    else:
+        # Show all commits
+        for commit in commits_list:
+            commit_num += 1
+            commit_hash, commit_title = commit.split(' ', 1)
+            sys.stdout.write('    ' + str(commit_num) + '. (\033[34m' +
+                commit_hash + '\033[0m) ' + commit_title)
+
+            if commit_num == 1:
+                sys.stdout.write("    \033[32m<-- old HEAD of '" + branch_name +
+                    "'\033[0m\n")
+            else:
+                sys.stdout.write('\n')
+
+    return num_commits
+
+# Set up command line arguments
+parser = argparse.ArgumentParser(usage='%(prog)s [ARGUMENTS]',
+    description='Set timestamps of all commits in a branch to the current time')
+parser.add_argument('-b', '--base', default='upstream/master',
+    help='branch (or git ref) to use as the base of the branch '
+        "(defaults to 'upstream/master')")
+args = vars(parser.parse_args())
+
+if not git.in_repo():
+    die("Not a git repository. Aborting")
+
+if not git.is_valid_commit(args['base']):
+    die("Invalid base '{}'. Aborting.".format(args['base']))
+
+branch_name = git.get_current_branch()
+if not branch_name:
+    die("No branch is currently checked out. Aborting.")
+
+num_commits = display_commits_list(args['base'], branch_name)
+
+# Ask for confirmation
+proceed = raw_input("Proceed (y/n)? ")
+if proceed != 'y' and proceed != 'Y':
+    die("Aborting.")
+
+if git.is_gpg_signing_enabled():
+    author_email = git.get_user_email()
+    if not author_email:
+        die("Failed to get the author email address. Aborting.")
+
+    gpg_sign_argument = '--gpg-sign=' + author_email
+else:
+    gpg_sign_argument = ''
+
+commits_range_argument = 'HEAD~' + str(num_commits)
+
+# Perform the actual resetting of the timestamps
+try:
+    if gpg_sign_argument:
+        git.run_command('rebase', '--ignore-date', gpg_sign_argument,
+            commits_range_argument)
+    else:
+        git.run_command('rebase', '--ignore-date', commits_range_argument)
```

Comment:
Silly, but you could accumulate the arguments in a list and then call `git.run_command(*args)`.